libs/libc/misc/lib_utsname.c

```diff
@@ -47,6 +47,20 @@
 #include <nuttx/version.h>
 #include <unistd.h>
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static char g_sysname[] = "NuttX";
+static char g_machine[] = CONFIG_ARCH;
+static char g_release[] = CONFIG_VERSION_STRING;
+
+#if defined(__DATE__) && defined(__TIME__)
+static char g_version[] = CONFIG_VERSION_BUILD " " __DATE__ " " __TIME__;
+#else
+static char g_version[] = CONFIG_VERSION_BUILD;
```

Comment:
```suggestion
static const char g_version[] = CONFIG_VERSION_BUILD;
```

Reply:
These data do not want to be read-only, they need to exist in memory, so they need to be located in the data segment. For example, if we want to match which elf all exported memory files correspond to, we need to read out the version and other information from the memory

Reply:
seems may be a special case? could we just change g_version to data and change other variables back to rodata?